Support int to datetime

sudo-plz opened the pull request:
Timestamp with int support.

Example
```
<?php

use Vcn\Pipette\Json;

$input = <<<JSON
{
  "time1": 123456
  "time2": "123476"
}
JSON;

try {
    $json = Json::parse($input);
    var_dump($json->field('time1')->timestamp());
    var_dump($json->field('time2')->dateTime('U'));
} catch (\Exception $e) {
    error_log($e);
}
```

Patch tag please.

Commit:
support int to datetime

Commit:
timestamp functions

rversteegt commented on src/Json/Value.php:
```diff
@@ -606,6 +606,49 @@ public function ¿dateTime(string $format = DateTime::ATOM): ?DateTimeImmutable
     }
 
+    /**
+     * Assert this value is an int, assert it conforms to a given date time format, then return that date time.
```
> Assert this value is a number, then return a date time measured in that many seconds since the Unix Epoch.

To remain consistent with the other documentation.

rversteegt commented on src/Json/Value.php:
```diff
@@ -606,6 +606,49 @@ public function ¿dateTime(string $format = DateTime::ATOM): ?DateTimeImmutable
+     * For timestamp strings, use dateTime('U').
+     *
+     * @return DateTimeImmutable
+     * @throws Exception\AssertionFailed If this value is not a string or int, or does not conform to the given format.
```
> If this value is not a number.

rversteegt commented on src/Json/Value.php:
```diff
@@ -606,6 +606,49 @@ public function ¿dateTime(string $format = DateTime::ATOM): ?DateTimeImmutable
+    public function timestamp(): DateTimeImmutable
+    {
+        $string = (string)$this->int();
+        $result = DateTimeImmutable::createFromFormat('U', $string);
```
```
$int = $this->int();

return new DateTimeImmutable("@$int");
```
This can't fail, and if it does, the failure is propagated from the constructor call.

rversteegt reviewed (changes requested):
`¿timestamp` should also be added to `OptionalValue`.

tests pls

Commit:
edited docs

Commit:
support PHP 8.1
sorry for the dev-main, phpspec doens´t have a tag with PHP 8.1 support yet but limited to a dev-dependency

Commit:
Remove dev-dependency from composer.json

Commit:
added tests

Commit:
100% coverage again
